lib/SILOptimizer/Transforms/PruneVTables.cpp

```diff
@@ -27,8 +27,13 @@ using namespace swift;
 
 namespace {
 class PruneVTables : public SILModuleTransform {
-  void runOnVTable(SILModule *M,
-                   SILVTable *vtable) {
+  void runOnVTable(SILModule *M, SILVTable *vtable) {
+    if (!vtable->getClass()->isOutermostPrivateOrFilePrivateScope()) {
```

Comment:
Turning this off for everything is not appropriate, we at least should be using the tools that we already have that handle this sort of thing. There are ways of checking this visibility while handling wmo/non-wmo that we should be using rather than referring to the ast.

Reply:
Reading SILVTable, it seems that there is a query that tells you if the vtable will be serialized or was serialized. Given that we seem to not have SILLinkage here, I imagine that is what one would want to use.

@eeckstein what are your thoughts?

Reply:
> Before this goes in I would like to understand more what is happening here.

This is a pretty bad regression impacting Linux and Windows builds.  Right now we incorrectly dispatch to the wrong function when the VWT is referenced outside of the module.

Reply:
Can you point me to what you have in mind for checking here?  I think that the WMO doesn't guarantee that every single reference is going to get fully monomorphized right?  But, sure WMO with internal visibility should allow additional cases.

Reply:
This problem can only occur in non-WMO mode or for "open" classes in WMO mode, right?
Can we disable this optimization just for "open" classes and for non-WMO mode (which is not so important)?